src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -51,83 +125,168 @@ public class ScriptTagHelper : TagHelper
-                    var encodedValue = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Value);
+                // Build the <script> tag that checks the test method and if it fails, renders the extra script.
+                builder.AppendLine();
+                builder.Append("<script>(")
+                   .Append(FallbackTestExpression)
+                   .Append("||");
 
-                    content.AppendFormat(CultureInfo.InvariantCulture, " {0}=\\\"{1}\\\"", encodedKey, encodedValue);
-                }
-                else
+                builder.Append("document.write(\"");
+
+                foreach (var src in fallbackSrcs)
                 {
-                    AppendSrc(content, attribute.Key);
+                    builder.Append("<script");
+
+                    if (!attributes.ContainsKey("src"))
+                    {
+                        AppendSrc(builder, "src", src);
+                    }
+
+                    foreach (var attribute in attributes)
+                    {
+                        if (!attribute.Key.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var encodedKey = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Key);
+                            var encodedValue = JavaScriptStringEncoder.Default.JavaScriptStringEncode(attribute.Value);
+
+                            builder.AppendFormat(
+                                CultureInfo.InvariantCulture,
+                                " {0}=\\\"{1}\\\"",
+                                encodedKey,
+                                encodedValue);
+                        }
+                        else
+                        {
+                            AppendSrc(builder, attribute.Key, src);
+                        }
+                    }
+
+                    builder.Append("><\\/script>");
                 }
+                
+                builder.Append("\"));</script>");
             }
+        }
 
-            content.Append("><\\/script>\"));</script>");
+        private void EnsureGlobbingUrlBuilder()
+        {
+            if (GlobbingUrlBuilder == null)
+            {
+                GlobbingUrlBuilder = new GlobbingUrlBuilder(
+                    HostingEnvironment.WebRootFileProvider,
+                    Cache,
+                    ViewContext.HttpContext.Request.PathBase);
+            }
+        }
+
+        private static void BuildScriptTag(
+            string content,
+            IDictionary<string, string> attributes,
+            StringBuilder builder)
+        {
+            builder.Append("<script");
+
+            foreach (var attribute in attributes)
+            {
+                builder.AppendFormat(CultureInfo.InvariantCulture, " {0}=\"{1}\"", attribute.Key, attribute.Value);
+            }
 
-            output.Content = content.ToString();
+            builder.Append(">");
```

Comment:
You chain elsewhere, should be consistent with these appends one after the other. (I prefer chaining, less noise).